cmd/zt_copy_blob_download_test.go

```diff
@@ -455,3 +455,211 @@ func (s *cmdIntegrationSuite) TestDownloadBlobContainerWithPattern(c *chk.C) {
 		c.Assert(strings.Contains(mockedRPC.transfers[0].Source[1:], common.AZCOPY_PATH_SEPARATOR_STRING), chk.Equals, false)
 	})
 }
+
+// test for include with one regular expression
+func (s *cmdIntegrationSuite) TestDownloadBlobContainerWithRegexInclude(c *chk.C) {
+	bsu := getBSU()
+
+	// set up the container with  blobs
+	containerURL, containerName := createNewContainer(c, bsu)
+	blobsToIgnore := scenarioHelper{}.generateCommonRemoteScenarioForBlob(c, containerURL, "")
+	defer deleteContainer(c, containerURL)
+	c.Assert(containerURL, chk.NotNil)
+	c.Assert(len(blobsToIgnore), chk.Not(chk.Equals), 0)
+
+	// add blobs that we wish to include
+	blobsToInclude := []string{"tessssssssssssst.txt", "subOne/tetingessssss.jpeg", "subOne/subTwo/tessssst.pdf"}
+	scenarioHelper{}.generateBlobsFromList(c, containerURL, blobsToInclude, blockBlobDefaultData)
+
+	// set up the destination with an empty folder
+	dstDirName := scenarioHelper{}.generateLocalDirectory(c)
+	defer os.RemoveAll(dstDirName)
+
+	// set up interceptor
+	mockedRPC := interceptor{}
+	Rpc = mockedRPC.intercept
+	mockedRPC.init()
+
+	// construct the raw input to simulate user input
+	rawContainerURLWithSAS := scenarioHelper{}.getRawContainerURLWithSAS(c, containerName)
+	rawContainerURLWithSAS.Path = path.Join(rawContainerURLWithSAS.Path, string([]byte{0x00}))
+	containerString := strings.ReplaceAll(rawContainerURLWithSAS.String(), "%00", "*")
+	raw := getDefaultCopyRawInput(containerString, dstDirName)
+	raw.recursive = true
+	raw.includeRegex = "es{4,}"
+
+ 	runCopyAndVerify(c, raw, func(err error) {
+		c.Assert(err, chk.IsNil)
+		// validate that the right number of transfers were scheduled
+		c.Assert(len(mockedRPC.transfers), chk.Equals, len(blobsToInclude))
```

Comment:
Is there a way to check that the actual values of the transfers match the values of blobsToInclude? 

Length is a bit unreliable.

Reply:
You could convert to maps and cross-check. I want to say there was a function for that SOMEWHERE in the test suite... But don't quote me on it.

Reply:
I was not able to find the specific function instead I am looping through each mockedRPC.transfers and checking if they are equal to blobsToInclude